File: frontend/src/services/officerService.js
import api from './api';

/**
 * Service to orchestrate municipal dispatches, task dispatches, and internal messaging.
 */

export const createOfficer = (name, email, password, department) => {
  return api.post('/api/officers', { name, email, password, department });
};

export const getAllOfficers = () => {
  return api.get('/api/officers');
};

export const assignIncident = (incidentId, officerId, deadline, priority, instructions) => {
  return api.post('/api/assignments', { incidentId, officerId, deadline, priority, instructions });
};

export const getAssignmentsForOfficer = () => {
  return api.get('/api/assignments/officer');
};

export const getAssignmentForIncident = (incidentId) => {
  return api.get(`/api/assignments/incident/${incidentId}`);
};

export const updateAssignmentStatus = (assignmentId, status, internalNotes = '', completionImageUrl = '', completionReport = '') => {
  return api.patch(`/api/assignments/${assignmentId}/status`, {
    status,
    internalNotes,
    completionImageUrl,
    completionReport
  });
};

export const sendChatMessage = (roomId, text) => {
  return api.post(`/api/chat/${roomId}`, { text });
};

export const getChatMessages = (roomId) => {
  return api.get(`/api/chat/${roomId}`);
};

File: backend/src/main/java/com/service/OfficerService.java
package com.service;

import com.model.Assignment;
import com.model.Message;
import com.model.Officer;
import java.util.List;

/**
 * Service orchestrating officer profiles, department task dispatches, and chats.
 */
public interface OfficerService {

    Officer createOfficer(String name, String email, String password, String department);

    List<Officer> getAllOfficers();

    Officer getOfficerByEmail(String email);

    void assignIncident(String incidentId, String officerId, Long deadline, String priority, String instructions);

    Assignment getAssignmentForIncident(String incidentId);

    List<Assignment> getAssignmentsForOfficer(String email);

    void updateAssignmentStatus(String assignmentId, String status, String internalNotes, String completionImageUrl, String completionReport);

    void sendInternalMessage(String chatRoomId, String text, String senderEmail);

    List<Message> getChatMessages(String chatRoomId);
}
